### PR TITLE
14.0 mig base substate

### PR DESCRIPTION
**Pull request by mourad-ehm**

### CODE REVIEW

File: base_substate/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from . import models
```
Comment (sebastienbeau) [inline base_substate/tests/__init__.py]: This will break your test, please read : https://github.com/akretion/odoo-test-helper
```suggestion
```

**Review by sebastienbeau — CHANGES_REQUESTED** [review]

**Comment by kittiu** [comment]
@mourad-ehm can you please squash your commits into 2 commits and a little nicer subject, please change to 
[14.0][MIG] base_substate.

Thanks!

File: base_substate/models/base_substate_mixin.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        return self.env["base.substate.type"].search(
+            [("model", "=", self._name)], limit=1
+        )
+
+    substate_id = fields.Many2one(
+        "base.substate",
+        string="Sub State",
+        ondelete="restrict",
+        default=lambda self: self._get_default_substate_id(),
+        track_visibility="onchange",
```
Comment (kittiu) [inline base_substate/models/base_substate_mixin.py:98]:
```suggestion
        tracking=True,
```

File: base_substate/views/base_substate_views.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        </field>
+    </record>
+    <record id="base_substate_view_form" model="ir.ui.view">
+        <field name="model">base.substate</field>
+        <field name="arch" type="xml">
+            <form string="Base Substate" name="base_substate">
+                <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <button
+                            name="toggle_active"
```
Comment (newtratip) [inline base_substate/views/base_substate_views.xml:24]: Please add web_ribbon Archived instead this
the example https://github.com/odoo/odoo/blob/14.0/addons/hr_holidays/views/hr_leave_type_views.xml#L38

**Comment by kittiu** [comment]
@mourad-ehm are you still working on this PR?

**Comment by kittiu** [comment]
@mourad-ehm I will make the new PR, ok?

**Comment by kittiu** [comment]
> @mourad-ehm I will make the new PR, ok?

@sebastienbeau do you know status of this PR?

**Comment by kittiu** [comment]
cc @mourad-ehm, @sebastienbeau
I create another PR to continue this work, https://github.com/OCA/server-ux/pull/366
Would you close this PR if you don't have time to continue, thank!

**Comment by sebastienbeau** [comment]
Yes go go go, thanks for your help

**Pull request closed without merging**